internal/keepers/service.go

```diff
@@ -306,67 +252,56 @@ func (s *onDemandUpkeepService) parallelCheck(ctx context.Context, keys []types.
 	return samples.Values(), nil
 }
 
-func (s *onDemandUpkeepService) aggregateWorkerResults(w *sync.WaitGroup, r *workerResults, sa *syncedArray[types.UpkeepResult], done chan struct{}) {
-	s.logger.Printf("starting service to read worker results")
-
-Outer:
-	for {
-		select {
-		case result := <-s.workers.results:
-			if result.Err == nil {
-				r.AddSuccess(1)
-
-				// Cache results
-				for i := range result.Data {
-					res := result.Data[i]
-					s.cache.Set(res.Key.String(), res, util.DefaultCacheExpiration)
-					if res.State == types.Eligible {
-						sa.Append(res)
-					}
-				}
-			} else {
-				r.SetLastErr(result.Err)
-				s.logger.Printf("error received from worker result: %s", result.Err)
-				r.AddFailure(1)
+func filterFromCache(keys []types.UpkeepKey, cache *util.Cache[types.UpkeepResult], samples *syncedArray[types.UpkeepResult]) ([]types.UpkeepKey, int) {
+	var keysToSend = make([]types.UpkeepKey, 0, len(keys))
+	var cacheHits int
+
+	for _, key := range keys {
+		// no RPC lookups need to be done if a result has already been cached
+		result, cached := cache.Get(key.String())
+		if cached {
+			cacheHits++
+			if result.State == types.Eligible {
+				samples.Append(result)
 			}
-			w.Done()
-		case <-done:
-			s.logger.Printf("done signal received for job result aggregator")
-			break Outer
+			continue
```

Comment:
🏄‍♂️ Personally I think I'd prefer an else here to append to the keysToSend rather than a continue to prevent it? But this is totally cool if you prefer the continue!

Reply:
This one I'm inconsistent on. According to official Go style, continue is preferred. But it's really just preference, right?